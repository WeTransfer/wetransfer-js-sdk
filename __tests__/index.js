const nock = require('nock');

const createWTClient = require('../src');

describe('createWTClient function', () => {
  beforeEach(() => {
    nock('https://dev.wetransfer.com')
      .post('/v1/authorize')
      .reply(200);
  });

  it('should return a client if API KEY is provided', async () => {
    try {
      await createWTClient();
    } catch ({ message }) {
      expect(message).toEqual('No API Key provided');
    }
  });

  it('should return an API client', async () => {
    const apiClient = await createWTClient('super-secret-api-key');
    expect(apiClient).toEqual({
      authorize: expect.any(Function),
      transfer: {
        addItems: expect.any(Function),
        completeFileUpload: expect.any(Function),
        create: expect.any(Function),
        uploadFile: expect.any(Function)
      }
    });
  });
});
