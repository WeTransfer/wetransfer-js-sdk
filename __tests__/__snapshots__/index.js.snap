// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createWTClient function should throw and error when no API KEY is provided 1`] = `[Error: No API Key provided]`;
